Robus/src/robus.c

```diff
@@ -467,3 +498,57 @@ void Robus_ShiftMaskCalculation(uint16_t ID, uint16_t ServiceNumber)
         ctx.IDMask[tempo / 8] |= 1 << ((tempo) % 8);
     }
 }
+
+/******************************************************************************
+ * @brief set node_connected variable
+ * @param state
+ * @return None
+ ******************************************************************************/
+void Robus_SetNodeDetected(network_state_t state)
+{
+    switch (state)
+    {
+        case NETWORK_LINK_DOWN:
+            node_connected.timeout_run = false;
+            node_connected.timeout     = 0;
```

Comment:
I think here you can consider `node_connected.timeout     == 0;` equivalent to your `node_connected.timeout_run = false;` this avoid the management of this extra variable.

Reply:
I think this could be error prone : what if LuosHAL_GetSystick() returns 0 in "node_connected.timeout     = LuosHAL_GetSystick();".

I think the solution proposed is safer with an extra cost which is not important.